packages/rules-engine/src/lib/runners/allIfFirst.spec.ts

```diff
@@ -0,0 +1,64 @@
+import { type Rule, type RuleContext } from "../..";
+import { appendOutput } from "../appendOutput";
+import { allIfFirst } from "./allIfFirst";
+
+interface Input {
+  a: number;
+  b: number;
+}
+
+type Output = number;
+
+type TestContext = RuleContext<Input, Output>;
+
+type TestRule = Rule<Input, Output>;
+
+const context: TestContext = {
+  input: { a: 1, b: 2 },
+  output: [],
+};
+
+const testRule1: TestRule = {
+  runIf: () => false,
+  run: (context) => appendOutput(context, 1),
+};
+
+const testRule2: TestRule = {
+  runIf: () => true,
+  run: (context) => appendOutput(context, 2),
+};
+
+const testRule3: TestRule = {
+  runIf: () => true,
+  run: (context) => appendOutput(context, 3),
+};
+
+const testRule4: TestRule = {
+  runIf: () => false,
+  run: (context) => appendOutput(context, 4),
+};
+
+describe("allIfFirst", () => {
+  describe("if", () => {
+    it("returns true if the first rule is true", () => {
+      expect(allIfFirst(testRule2, testRule1).runIf(context.input)).toBe(true);
+    });
+
+    it("returns false if the first rule is false", () => {
```

Comment:
Should we add a test for if an empty array is passed?

Reply:
Done, good idea.